olive/evaluator/metric_backend.py

```diff
@@ -0,0 +1,91 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+from abc import abstractmethod
+from typing import Any, Dict, Union
+
+from olive.common.auto_config import AutoConfigClass, ConfigBase
+from olive.common.config_utils import ConfigParam
+from olive.evaluator.accuracy import AccuracyBase
+from olive.evaluator.metric import MetricResult, SubMetricResult
+
+
+class MetricBackend(AutoConfigClass):
+    registry: Dict[str, "MetricBackend"] = {}
+
+    def __init__(self, config: Union[ConfigBase, Dict[str, Any]] = None) -> None:
+        super().__init__(config)
+
+    @staticmethod
+    def _default_config() -> Dict[str, ConfigParam]:
+        return {}
+
+    @abstractmethod
+    def measure(self, preds, targets, sub_metric) -> SubMetricResult:
+        raise NotImplementedError()
+
+    def combine_measure(self, preds, targets, metrics) -> MetricResult:
```

Comment:
>metrics [](http://example.com/codeflow?start=46&length=7)

add type notation for metrics and sub_metric?

Reply:
Added.